File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        StringBuilder b = new StringBuilder(); 
        
        for(int x =0; x <s.length(); x++){
            if(Character.isLetterOrDigit(s.charAt(x))){
                b.append(s.charAt(x));
            }
        }
        return recursively(b,0,b.length());        
    }    
    public boolean recursively(StringBuilder b,int i,int n){      
      if(i>= n/2) return true;          
          if(b.charAt(i) != b.charAt(n-i-1)){
              return false;
          }
          return recursively(b,i+1,n);
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.cs
public class Solution {
    private static bool IsAlphanumeric(char x) {
        return (x >= 'a' && x <= 'z') || (x >= 'A' && x <= 'Z') || (x >= '0' && x <= '9');
    }
    public bool IsPalindrome(string s) {
        int left = 0;
        int right = s.Length - 1;

        while (left < right) {
            while (left < right && !IsAlphanumeric(s[left])) left++;
            while (left < right && !IsAlphanumeric(s[right])) right--;
            if (char.ToLower(s[left++]) != char.ToLower(s[right--])) return false;
        }
        return true;
    }
}
